### DISCUSSION TITLE
How to list used keys?

### DISCUSSION THREAD

**Discussion by rostero1** [TanStack/query#5589] [Q&A]
Is there an API for listing keys?  For example, let's say I have `["team", "players", number]`.  How do I get all the player numbers?  I'd use this info to pass to queries.

```
const results = useQueries({
  queries: ids.map(id => [
    { queryKey: ['team', 'players', id], queryFn: () => fetchPlayer(id), staleTime: Infinity },
  ]),
})
```

I could keep this in React useState somewhere.

**Comment by TkDodo** [comment]
keys should be computable from their input (id in your case). If you have the ids, you should be able to build the keys at all times. Have a read here: https://tkdodo.eu/blog/effective-react-query-keys
